### PR TITLE
PERF: Only do case sensitive check when not lower case

### LINKED ISSUE #12876: Pandas IO to_sql extremely slow when checking for potentially case sensitivity issues
This is related to #7815

Since this fix, when checking for case sensitivity issues for MySQL using InnoDB engine with large numbers of tables, Class SQLDatabase.to_sql can take a long time (I currently have ~30,000 tables and this call takes ~10 seconds).
This is due to the 'SHOW FULL TABLES' call coming from SQLAlchemy, from these lines of code specifically
```python 
engine = self.connectable.engine
with self.connectable.connect() as conn:
    table_names = engine.table_names(
        schema=schema or self.meta.schema,
        connection=conn,
     )
```
A suggested fix would be to add a parameter to the ``to_sql`` function.
The parameter might look something like, ``case_check='warn'``
and the fix might be something like

```python 
if case_check == 'raise' and not name.islower():
    raise Exception("table name must be lower case when case_check = 'raise'")
current_code()
if case_check == 'warn':
    engine = self.connectable.engine
    with self.connectable.connect() as conn:
        table_names = engine.table_names(
            schema=schema or self.meta.schema,
            connection=conn,
         )
```

Versions:

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.6.final.0
python-bits: 64
OS: Linux
OS-release: 3.13.0-83-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_IE.UTF-8

pandas: 0.18.0
nose: 1.3.7
pip: 8.1.1
setuptools: 18.2
Cython: 0.23.4
numpy: 1.10.1
scipy: 0.16.1
statsmodels: None
xarray: None
IPython: None
sphinx: 1.3.1
patsy: None
dateutil: 2.4.2
pytz: 2015.7
blosc: None
bottleneck: None
tables: 3.2.0
numexpr: 2.4.6
matplotlib: 1.5.0
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.9
pymysql: None
psycopg2: None
jinja2: 2.8
boto: 2.38.0

### PR DESCRIPTION
**Pull request by RogerThomas**
- [x] closes #12876
- [ ] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.18.1.txt [hunk 1]
```diff
@@ -203,6 +203,7 @@ Performance Improvements
 
 
 
+- Improved performance of ``DataFrame.to_sql`` when checking case sensitivity. Now only checks if table has been created correctly when table name is not lower case. (:issue:`12876`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.1.txt]: case sensitivity for tables.

File: pandas/io/sql.py [hunk 2]
```diff
@@ -1248,19 +1248,23 @@ def to_sql(self, frame, name, if_exists='fail', index=True,
...
-                          "as such in the database after writing the table, "
-                          "possibly due to case sensitivity issues. Consider "
-                          "using lower case table names.".format(name),
-                          UserWarning)
+        if (not name.isdigit() and not name.islower()):
+            # check for potentially case sensitivity issues (GH7815)
+            # Only check when name is not a number and name is not lower case
+            engine = self.connectable.engine
+            with self.connectable.connect() as conn:
+                table_names = engine.table_names(
```
Comment (jreback) [inline pandas/io/sql.py]: can you give some tests for this (and for the warning).

Comment (RogerThomas) [inline pandas/io/sql.py]: Sure will do

Comment (RogerThomas) [inline pandas/io/sql.py]: @jreback Apparently the warning itself can't be tested
https://github.com/pydata/pandas/blob/master/pandas/io/tests/test_sql.py#L919
I can add a test to ensure the warning is not triggered when using a number as a table name though?

Comment (jreback) [inline pandas/io/sql.py]: hmm @jorisvandenbossche

Comment (jorisvandenbossche) [inline pandas/io/sql.py]: Well, I indeed put a comment here a while ago that I couldn't test it .. :-) 
And I still think this is the case, as we would have to change the settings of the postgresql table on travis (to have this situation occurring), and not sure how to do this/if this is worth the trouble

Comment (jorisvandenbossche) [inline pandas/io/sql.py]: To conclude: I think it is OK to merge this without an explicit test

**Comment by jreback** [comment]
thanks @RogerThomas

**Pull request closed without merging**